File: Expressions/budget.c
#include <stdio.h>

int main(void){
    float income, rent, utilities, groceries, transportation, savings, expenses, spend;
    float prent, putilities, pgroceries, ptransportation, psavings, pexpenses;
    printf("This is a budger calculator.\n How much do you make a month?\n");
    scanf("%f", &income);
    printf("How much your rent cost?\n");
    scanf("%f", &rent);
    printf("How much your utilities cost?\n");
    scanf("%f", &utilities);
    printf("How much do your grocieries cost?\n");
    scanf("%f", &groceries);
    printf("How much your transportation cost?\n");
    scanf("%f", &transportation);
    savings = income * .2; 
    expenses = rent + utilities + groceries + transportation;
    spend = income - expenses - savings;
    printf("You make $%.2f\n", income);
    printf("Your expenses are $%.2f\n", expenses);
    printf("Your savings are $%.2f\n", savings);
    printf("Your spending money is $%.2f\n", spend);
    prent = rent/income * 100;
    putilities = utilities/income * 100;
    pgroceries = groceries/income * 100;
    ptransportation = transportation/income * 100;
    psavings = savings/income * 100;
    pexpenses = expenses/income * 100;
    printf("Your rent is %d%% of your income.\n", (int) prent);
    printf("Your utilities are %d%% of your income.\n", (int) putilities);
    printf("Your groceries are %d%% of your income.\n", (int) pgroceries);
    printf("Your transportation costs are %d%% of your income.\n", (int) ptransportation);
    printf("Your savings are %d%% of your income.\n", (int) psavings);
    printf("Your expenses are %d%% of your income.\n", (int) pexpenses);
    return 0;
}

File: Expressions/budget.py
def main():
    income = float(input("Enter your monthly income: $"))

    rent = float(input("How much is your rent cost: $"))
    utilities = float(input("How much are your utilities cost: $"))
    groceries = float(input("How much are your groceries cost: $"))
    transportation = float(input("How much is your transportation cost: $"))

    total_expenses = rent + utilities + groceries + transportation

    savings = income * 0.20
    remaining_amount = income - total_expenses - savings

    print("\nFinancial Summary:")
    print(f"Monthly Income: ${income:.2f}")
    print(f"Total Expenses: ${total_expenses:.2f}")
    print(f"Suggested Savings (20% of Income): ${savings:.2f}")
    print(f"Amount Left to Spend: ${remaining_amount:.2f}")

    rent_percentage = (rent / income) * 100
    utilities_percentage = (utilities / income) * 100
    groceries_percentage = (groceries / income) * 100
    transportation_percentage = (transportation / income) * 100
    savings_percentage = (savings / income) * 100

    print(f"\nPercentage of Income:")
    print(f"Rent/Mortgage: {rent_percentage:.2f}%")
    print(f"Utilities: {utilities_percentage:.2f}%")
    print(f"Groceries: {groceries_percentage:.2f}%")
    print(f"Transportation: {transportation_percentage:.2f}%")
    print(f"Savings: {savings_percentage:.2f}%")

if __name__ == "__main__":
    main()